File: frontend/demo/component/textfield/text-field-text-alignment.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/text-field';
import '@vaadin/form-layout';
import '@vaadin/form-layout/vaadin-form-item.js';
import type { FormLayoutResponsiveStep } from '@vaadin/form-layout';
import { applyTheme } from 'Frontend/generated/theme';

const layoutSteps: FormLayoutResponsiveStep[] = [
  {
    minWidth: 0,
    columns: 1,
    labelsPosition: 'aside',
  },
];

@customElement('text-field-text-alignment')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-form-layout .responsiveSteps="${layoutSteps}">
        <vaadin-form-item>
          <label slot="label">Left</label>
          <vaadin-text-field value="value"></vaadin-text-field>
        </vaadin-form-item>

        <vaadin-form-item>
          <label slot="label">Center</label>
          <vaadin-text-field value="value" theme="align-center"></vaadin-text-field>
        </vaadin-form-item>

        <vaadin-form-item>
          <label slot="label">Right</label>
          <!-- tag::snippet[] -->
          <vaadin-text-field value="value" theme="align-right"></vaadin-text-field>
          <!-- end::snippet[] -->
        </vaadin-form-item>
      </vaadin-form-layout>
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/textfield/TextFieldTextAlignment.java
package com.vaadin.demo.component.textfield;

import com.vaadin.flow.component.formlayout.FormLayout;
import com.vaadin.flow.component.formlayout.FormLayout.ResponsiveStep.LabelsPosition;
import com.vaadin.flow.component.textfield.TextField;
import com.vaadin.flow.component.textfield.TextFieldVariant;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("text-field-text-alignment")
public class TextFieldTextAlignment extends FormLayout {

    public TextFieldTextAlignment() {
        setResponsiveSteps(new ResponsiveStep("0" , 1, LabelsPosition.ASIDE));

        TextField left = new TextField();
        left.setValue("value");
        addFormItem(left, "Left");

        TextField center = new TextField();
        center.setValue("value");
        center.addThemeVariants(TextFieldVariant.LUMO_ALIGN_CENTER);
        addFormItem(center, "Center");

        // tag::snippet[]
        TextField right = new TextField();
        right.setValue("value");
        right.addThemeVariants(TextFieldVariant.LUMO_ALIGN_RIGHT);
        // end::snippet[]
        addFormItem(right, "Right");
    }

    public static class Exporter extends DemoExporter<TextFieldTextAlignment> { // hidden-source-line
    } // hidden-source-line
}
